articles/101_build_tool.md

```diff
@@ -387,16 +387,26 @@ For option **A)** the follow items should be considered:
 - Update code base to Python 3.5+.
 - Refactor code base to reduce coupling (e.g. separate [API](https://github.com/catkin/catkin_tools/blob/2cae17f8f32b0193384d2c7734afee1c60c4add2/catkin_tools/execution/controllers.py#L183-L205) for output handling).
 - Additional functionality to build Gazebo including its dependencies.
+- Whether or not to continue supporting the *devel space*.
 
 For option **B)** the follow items should be considered:
 
 - Support cross compilation.
 - Support `DESTDIR`.
 - Support a feature similar to the `profile` verb of `catkin_tools`.
+- Support for a shared GNU Make job sever.
+- Support for `GNUInstallDirs`
+  - Not sure about the status of this, it would be in `colcon`'s generated shell files if anywhere.
+  - Should have a test for this case.
+- Test for, and fix if necessary, correct topological order with dependencies across workspaces.
+  - See: [https://github.com/ros/catkin/pull/590](https://github.com/ros/catkin/pull/590)
 
 ## Summary and Decision
 
-Based on the above information a decision has been made to pick `colcon` as the universal build tool which was not an easy one.
+Based on the above information a decision has been made to pick `colcon` as the universal build tool.
+
+The decision was made after considering the input of ROS 2 team members and some ROS 1 users.
```

Comment:
What happened to consulting ROS 2 users?

Reply:
And "some ROS1 users" is terribly unspecific. ROS1 has been working by REPs, and OSRF religiously decreed "ROS1 will not be disrupted". So now you want to disrupt ROS1 without a REP?

Reply:
> What happened to consulting ROS 2 users?

I am not sure what kind of feedback we would expect from this. `colcon` basically works the same way as `ament_tools`. Some of the command line options are named slightly different. If that is a major concern it would be easy to provide an executable named `ament` which provides the exact same CLI (I have a prototype of that locally from when I was exploring such options). But since the difference is really small (see http://colcon.readthedocs.io/en/latest/migration/ament_tools.html) I am not sure if that would even be valuable or more confusing.

Maybe you can clarify what other kind of feedback you would expect?

Reply:
> What happened to consulting ROS 2 users?

I think feedback on this design document update _is_ the solicitation for feedback from ROS 2 users and others. Though I don't know if it would impact the decision that's been made one way or the other.

> And "some ROS1 users" is terribly unspecific.

That was intentional, because I didn't take time to ask if I could mention them directly, also I didn't feel it was material to the point.

> So now you want to disrupt ROS1 without a REP?

Well, there isn't a REP to update or replace w.r.t. the build tool in ROS. It's more of a recommendation than a requirement, and not the same in my mind as the switch from `rosbuild` to `catkin`.

Other things which would actually be disruptive (require changes in existing code) would require REP's, like getting rid of the devel space, and deciding on whether or not to pursue those or not is apparently also blocking roll out of the universal build tool in ROS 1, so I don't it will be disruption without REP's personally, but I suppose we'll have to see.

To be fair, had we decided to use catkin_tools, roll out in ROS 1 would have just been fix a few final things and change documentation to recommend catkin_tools instead of catkin_make, but there would not have been a need for a REP in that case, in my opinion.

Reply:
I don't think there is much feedback to provide from the point of view of a ROS 2 user. I'm actually in favour of the change in tool, especially because it is so non-disruptive. But I'm disappointed that I only get to provide that opinion after the decision has been made. I find it annoying that the OSRF assumes that existing ROS 2 users will have no valid or useful input into a high level decision such as this. We went from a state we were in a year ago to suddenly having a new tool and a decision made to use it without any notice.

Reply:
Yeah, I agree that community feedback before a decision would be nice, but as a member of the team I also understand the desire to get things moving. I’m certain your feedback is considered but it would have to be a significant amount of negative feedback to change the direction. And that was the case before the decision was made as well.